### DISCUSSION TITLE
Thank you for this library!

### DISCUSSION THREAD

**Discussion by jetersen** [mjebrahimi/EasyCompressor#77] [General]
After spending way too much time trying to get compression on a stream to work together with an tar archive, your library with dependency injection made it a easy to trust the compression was working correctly.

Finally found out that instead of trying to use SharpZipLib or SharpCompress to create the tar... I could simply use ZipArchive from C# core to properly wrangle my needs.

@mjebrahimi once again, Kudos for this excellent library!

**Comment by mjebrahimi** [comment]
Hi @jetersen 
Thank you for your interest.
Zip supports these [compression methods](https://en.wikipedia.org/wiki/ZIP_(file_format)#Compression_methods): `Store (no compression)`, `Deflate`, `Deflate64`, `BZip2`, `LZMA`, `PPMd`, ...
C# `ZipFile`, `ZipArchive`, `ZipArchiveEntry` classes by default use the **Deflate** compression method.
You can use **EasyCompressor** for `Deflate`, `GZip`, and `LZMA` compression methods.
But if you want other compression methods, you can use SharpZipLib ([supported methods](https://icsharpcode.github.io/SharpZipLib/help/api/ICSharpCode.SharpZipLib.Zip.CompressionMethod.html)) and SharpCompress ([supported methods](https://github.com/adamhathcock/sharpcompress/blob/master/FORMATS.md#supported-format-table)).

**Comment by AlyxSharkBite** [comment]
I agree with @jetersen  this library is really handy.  Love the DI Implementation on it!
